ipld/amt/src/amt.rs

```diff
@@ -37,32 +44,31 @@ use crate::{
 /// // Generate cid by calling flush to remove cache
 /// let cid = amt.flush().unwrap();
 /// ```
-#[derive(Debug)]
-pub struct Amt<V, BS> {
-    root: Root<V>,
-    block_store: BS,
-}
+pub type Amt<V, BS> = AmtImpl<V, BS, V3>;
+/// Legacy amt V0
+pub type Amtv0<V, BS> = AmtImpl<V, BS, V0>;
 
-impl<V: PartialEq, BS: Blockstore> PartialEq for Amt<V, BS> {
+impl<V: PartialEq, BS: Blockstore, Ver: PartialEq> PartialEq for AmtImpl<V, BS, Ver> {
```

Comment:
We shouldn't _need_ PartialEq for `Ver`, right?

Reply:
It shouldn't, I too found it weird but here's the error:

```
error[E0369]: binary operation `==` cannot be applied to type `RootImpl<V, Ver>`
  --> ipld/amt/src/amt.rs:53:19
   |
53 |         self.root == other.root
   |         --------- ^^ ---------- RootImpl<V, Ver>
   |         |
   |         RootImpl<V, Ver>
   |
help: consider restricting type parameter `Ver`
   |
51 | impl<V: PartialEq, BS: Blockstore, Ver: std::cmp::PartialEq> PartialEq for AmtImpl<V, BS, Ver> {
```

Reply:
Hm. Ok, so, I think that's just a derive issue (you could manually implement it). But honestly, it doesn't really matter.